src/v_video.c

```diff
@@ -858,6 +858,29 @@ void V_DrawHorizLine(int x, int y, int scrn, int width, byte color)
   }
 }
 
+int screenshade;
+void V_ShadeScreen(void)
+{
+  int y;
+  byte *dest = screens[0];
+  const int targshade = 20;
+  static int oldtic;
+  
+  for (y = 0; y < (SCREENWIDTH << hires) * (SCREENHEIGHT << hires); y++)
```

Comment:
Okay, my idea is this: If shading is enabled, this function is called at least once during each tic. If `oldtic` is more than one tic away, `screenshade` should get reset.  Also, `screenshade` should be made static and get restricted to the scope of the function.

Reply:
Could you please apply this and try if it works for you?
```patch
diff --git a/src/d_main.c b/src/d_main.c
index 70a9bd4..6bc72db 100644
--- a/src/d_main.c
+++ b/src/d_main.c
@@ -224,10 +224,6 @@ void D_Display (void)
   extern boolean setup_active;
   extern int menu_background;
 
-  if (!((automapactive && automapoverlay == 2)
-        || (setup_active && menu_background == 2)))
-    screenshade = 0;
-
   if (demobar && PLAYBACK_SKIP)
   {
     if (HU_DemoProgressBar(false))
diff --git a/src/v_video.c b/src/v_video.c
index dc6df7f..5b97718 100644
--- a/src/v_video.c
+++ b/src/v_video.c
@@ -858,14 +858,20 @@ void V_DrawHorizLine(int x, int y, int scrn, int width, byte color)
   }
 }
 
-int screenshade;
 void V_ShadeScreen(void)
 {
   int y;
   byte *dest = screens[0];
   const int targshade = 20;
-  static int oldtic;
-  
+  static int oldtic = -1;
+  static int screenshade;
+
+  // [FG] longer than one tic ago? start a new sequence
+  if (gametic - oldtic > 1)
+  {
+    screenshade = 0;
+  }
+
   for (y = 0; y < (SCREENWIDTH << hires) * (SCREENHEIGHT << hires); y++)
   {
     dest[y] = colormaps[0][screenshade * 256 + dest[y]];
@@ -874,11 +880,12 @@ void V_ShadeScreen(void)
   if (screenshade < targshade && gametic != oldtic)
   {
     screenshade += 2;
-    oldtic = gametic;
 
     if (screenshade > targshade)
       screenshade = targshade;
   }
+
+  oldtic = gametic;
 }
 
 //
diff --git a/src/v_video.h b/src/v_video.h
index ececc43..4478fd9 100644
--- a/src/v_video.h
+++ b/src/v_video.h
@@ -135,7 +135,6 @@ void V_PutBlock(int x, int y, int scrn, int width, int height, byte *src);
 
 void V_DrawHorizLine(int x, int y, int scrn, int width, byte color);
 
-extern int screenshade;
 void V_ShadeScreen(void);
 
 // [FG] colored blood and gibs
```

Reply:
It works, yes. Not sure if I should've asked before implementing it in the latest commit, though, so I won't mark this as resolved myself.